ddl/column.go

```diff
@@ -308,6 +332,52 @@ func doModifyColumn(t *meta.Meta, job *model.Job, newCol *model.ColumnInfo, oldN
 	// }
 	// }
 
+	// Get sessionctx from context resource pool.
+	var ctx sessionctx.Context
+	ctx, err = w.sessPool.get()
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+	defer w.sessPool.put(ctx)
+
+	if !mysql.HasNotNullFlag(oldCol.Flag) && mysql.HasNotNullFlag(newCol.Flag) {
+		err = CheckForNullValue(ctx, dbInfo.Name, tblInfo.Name, oldCol.Name, newCol.Name)
+		if err != nil {
+			job.State = model.JobStateCancelled
+			return ver, errors.Trace(err)
+		}
+
+		if !mysql.HasPreventNullInsertFlag(oldCol.Flag) {
+			// Prevent this field from inserting null values.
+			tblInfo.Columns[oldCol.Offset].Flag |= mysql.PreventNullInsertFlag
+			ver, err = updateVersionAndTableInfo(t, job, tblInfo, true)
+			if err != nil {
+				return ver, errors.Trace(err)
+			}
+		} else {
+			// Modify the type defined Flag to NotNullFlag.
+			tblInfo.Columns[oldCol.Offset].Flag = newCol.Flag
```

Comment:
I think only modify the `Flag to NotNullFlag` and clear the `PreventNullInsertFlag` maybe better.
```go
tblInfo.Columns[oldCol.Offset].Flag |= mysql.NotNullFlag
tblInfo.Columns[oldCol.Offset].Flag = oldCol.Flag &^ mysql.PreventNullInsertFlag
```